src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -143,18 +155,29 @@ def load_command_table(self, command_loader):
                 _parser=command_parser)
 
     def validation_error(self, message):
-        telemetry.set_user_fault('validation error: {}'.format(message))
-        return super(AzCliCommandParser, self).error(message)
+        az_error = AzCLIError(AzCLIErrorType.ValidationError, message, command=self.prog)
+        az_error.print_error()
+        az_error.send_telemetry()
+        self.exit(2)
 
     def error(self, message):
-        telemetry.set_user_fault('parse error: {}'.format(message))
-        args = {'prog': self.prog, 'message': message}
-        with CommandLoggerContext(logger):
-            logger.error('%(prog)s: error: %(message)s', args)
-        self.print_usage(sys.stderr)
-        # Manual recommendations
-        self._set_manual_recommendations(args['message'])
-        # AI recommendations
+        # Get a recommended command from the CommandRecommender
+        command_arguments = self._get_failure_recovery_arguments()
+        recommender = CommandRecommender(*command_arguments, self.cli_ctx)
+        recommender.set_help_examples(self.get_examples(self.prog))
+        recommendation = recommender.recommend_a_command()
```

Comment:
We'd better make a switch to allow the user to turn it off, especially when core is used by other CLI tools like Azure Sphere CLI or used in an air-gapped cloud.

Something like:

```
az config set core.error_recommendation=False
```

Reply:
client configuration may not help in this case. This is quite advanced behavior. Behavior of huge # of installation/users isn't controllable. better azure cli be more defensive in calling into aladdin. 
- set call timeout. no retry
- aladdin call failure will not block cli code flow
- not call aladdin in clouds in air-gapped clouds. hi @mirdaki, is aladdin service accessible from other public clouds like German, Mookcake?

Reply:
Hi @yungezz, as far as I understand, if they can reach out to the public internet, they should be able to use the Aladdin service. I will add there is already a check for the air-gapped clouds, made in this [commit](https://github.com/Azure/azure-cli/pull/14855/commits/06888502590f263b34b6d91627d98dd0aedd6259).

I'm all for giving users more control of their client, but you know there behavior better than I do to know if this is a config item they would use.

Reply:
- Air-gapped cloud now is not allowed to call the api 
- Service is using hashed UserID to stop repeated api calls

So, let's just keep the current status. We can disscuss whether to add a config in the future if the are new cases.